records/validation.go

```diff
@@ -0,0 +1,49 @@
+package records
+
+import (
+	"fmt"
+	"net"
+)
+
+// validateMasters checks that each master in the list is a properly formatted host:ip pair.
+// duplicate masters in the list are not allowed.
+// returns nil if the masters list if empty, or else all masters in the list are valid.
+func validateMasters(ms []string) error {
+	valid := map[string]struct{}{}
```

Comment:
Since you know the upper bound of the size of the set, you can pre-allocate it. Same as below.
```go
valid := make(map[string]struct{}, len(ms))
```